### PR TITLE
add pad style: zeropadding for function pad & unpad

### PR DESCRIPTION
**Pull request by RIP-GitHub**
Add zeropadding for DES/AES encrypt

### CODE REVIEW

**Review by Varbin — COMMENTED** [review]
I do not like zero padding - it disallows null bytes at the end of messages, which might be problematic. As people often will just use what is available in libraries (even if there are warnings are included), I am not sure if it is a good idea to include zero padding in this library.

To avoid misunderstandings: This is only my personal opinion.

File: lib/Crypto/Util/Padding.py [hunk 1]
```diff
@@ -59,6 +59,8 @@ def pad(data_to_pad, block_size, style='pkcs7'):
         padding = bchr(0)*(padding_len-1) + bchr(padding_len)
     elif style == 'iso7816':
         padding = bchr(128) + bchr(0)*(padding_len-1)
+    elif style == 'zero':
+        padding = bchr(0)*padding_len
```
[the remark this answers is not in the document]
Comment (RIP-GitHub) [reply lib/Crypto/Util/Padding.py:63]: I think you're right. If the byte string to be filled ends with zero, you can't use zeropading
```
elif style == 'zero':
    padding = bchr(0) * padding_len
    if len(data_to_pad)!=0 and data_to_pad[-1]==bord(0):
        raise ValueError("Use zero padding when data to pad can't ends with \\x00")
```

File: lib/Crypto/Util/Padding.py [hunk 2]
```diff
@@ -102,6 +104,10 @@ def unpad(padded_data, block_size, style='pkcs7'):
             raise ValueError("Padding is incorrect.")
         if padding_len>1 and padded_data[1-padding_len:]!=bchr(0)*(padding_len-1):
             raise ValueError("ISO 7816-4 padding is incorrect.")
+    elif style == 'zero':
+        padding_len = pdata_len - padded_data.find(bchr(0))
```
[the remark this answers is not in the document]
Comment (RIP-GitHub) [reply lib/Crypto/Util/Padding.py:108]: I do not like zero padding either, after I write code of unpad for zeropadding.
in `unpad(b"AB\x00CD\x00\x00\x00", 8, "zero")`, we know padding length is 3. I can not find the better way to get padding length
```
padding_len = 0
reverse_last_block = padded_data[-block_size:][::-1]
if reverse_last_block == bchr(0)*block_size:
    padding_len = block_size
else:
    for index, item in enumerate(reverse_last_block):
        if bchr(item)==bchr(0) and bchr(reverse_last_block[index+1])!=bchr(0):
            padding_len = index + 1
            break
```

File: lib/Crypto/Util/Padding.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Varbin) [reply lib/Crypto/Util/Padding.py:63]: `data_to_pad[-1] == bord(0)` --> Shouldn't this be `bord(data_to_pad[-1]) != 0` instead?

[the remark this answers is not in the document]
Comment (RIP-GitHub) [reply lib/Crypto/Util/Padding.py:63]: It is a good suggestion

File: lib/Crypto/Util/Padding.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Varbin) [reply lib/Crypto/Util/Padding.py:108]: My suggestion of using `.rfind` was rubbish. 

If would have written it as:

```python3
position = 0
reversed_last_block = padded_data[-block_size:][::-1]
while bord(reversed_last_block[position]) == 0 and position < block_size:
  position += 1
padding_len = position
```

[the remark this answers is not in the document]
Comment (RIP-GitHub) [reply lib/Crypto/Util/Padding.py:108]: maybe `while position < block_size and bord(reversed_last_block[position]) == 0:` better, it can avoid IndexError

**Comment by Legrandin** [comment]
Sorry, I do not think that "zero padding" is a good addition. Any of the existing padding styles are preferable.

**Pull request closed without merging**